c10/cuda/CUDAStream.cpp

```diff
@@ -163,6 +163,11 @@ static void initGlobalStreamState() {
   int leastPriority = -1, greatestPriority = -1;
   C10_CUDA_CHECK(
       cudaDeviceGetStreamPriorityRange(&leastPriority, &greatestPriority));
+  // Note [HIP stream priorities]
+  // HIP stream priorities are 1=low, 0=default, -1=high which differs from CUDA
+  // which is 0=default, -1=high, -2=higher etc.
+  // Clamp leastPriority to 0 for HIP.
+  leastPriority = 0;
```

Comment:
Shouldn't this be guarded with `USE_ROCM`?

Reply:
Also, are you saying that `cudaDeviceGetStreamPriorityRange` is not clamping priorities correctly on ROCM?

Reply:
Yeah, I debated needing an ifdef there.  cuda returns 0 for leastPriority, so it's a no-op for cuda.  hip returns 1.  it's a behavior difference between HIP and CUDA (see Note I added in comments in the PR diff).  HIP and CUDA agree that 0 is the default stream priority.  but HIP adds a lower priority value of 1 in addition to the higher priority value of -1.  This change here is to make HIP behave in accordance with the semantics of pytorch and cuda where 0 is the lowest AND default priority.  We're basically ignoring the fact that HIP provides a lower priority than 0.  Later in the code, the priority range is iterated over and without this change HIP would end up creating 3 stream pools for priorities 0, -1, and another -1.

I don't expect CUDA will ever have leastPriority NOT be 0, and there is a TORCH_ASSERT elsewhere in the code to make sure it never changes.  Do you still want this #ifdef protected for USE_ROCM?

Reply:
I would prefer to keep an ifdef here and do not assume what `cudaDeviceGetStreamPriorityRange` returns/might return in the next version of CUDA